### PR TITLE
CHET-343: Plugin smoketest automation

### PR DESCRIPTION
**Pull request by tarka**
This runs a Postgres server, a production Jira image (from docker hub), and a basic Cypress test suite. 

This is an interim PR with the core functionality. The test suite is currently limited to checking the plugin has loaded, and is missing documentation updates. These will be added in another PR.

### CODE REVIEW

File: jira-e2e-tests/docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+      - 'POSTGRES_PASSWORD=password'
+      - 'POSTGRES_ENCODING=UNICODE'
+      - 'POSTGRES_COLLATE=C'
+      - 'POSTGRES_COLLATE_TYPE=C'
+      - "LANG=C"
+
+  jira:
+    build:
+      context: ./jira
+    #image: atlassian/jira-software
```
Comment (bpar476) [inline jira-e2e-tests/docker-compose.yml:20]: Why is this commented out?

**Comment by bpar476** [comment]
Can we cache the docker repository? 🤔

File: jira-e2e-tests/docker-compose.yml (hunk 1 above)
Comment (tarka) [reply jira-e2e-tests/docker-compose.yml:20]: Old code; I'll remove it.Turns out we have to build our own to include the necessary 'wait for postgres' functionality.

**Comment by tarka** [comment]
> Can we cache the docker repository? thinking

Not easily; it's an [outstanding feature request](https://github.com/actions/cache/issues/31). There's ways of doing it but they're a bit hacky. In practice the Docker downloads are fast enough; I suspect they proxy the requests to a local cache.

**Pull request merged**